manifests/init.pp

```diff
@@ -34,53 +34,98 @@
 #   }
 #
 class cron (
-  $manage_package = true,
-  $manage_service = true,
-  $service_ensure = 'running',
-  $service_enable = true,
-  $service_name   = $::cron::params::service_name,
-  $package_ensure = 'installed',
-  $package_name   = $::cron::params::package_name,
-) inherits cron::params {
+  String[1]      $service_name,
+  String[1]      $package_name,
+  Boolean        $manage_package = true,
+  Boolean        $manage_service = true,
+  Variant[
+    Boolean,
+    Enum[
+      'running',
+      'stopped',
+    ]
+  ]              $service_ensure = 'running',
+  Variant[
+    Boolean,
+    Enum[
+      'manual',
+      'mask',
+    ]
+  ]              $service_enable = true,
+  String[1]      $package_ensure = 'installed',
+  Array[String]  $users_allow    = [],
+  Array[String]  $users_deny     = [],
+  Boolean        $manage_users_allow = false,
+  Boolean        $manage_users_deny  = false,
+) {
 
-  validate_bool($manage_package, $manage_service, $service_enable)
+  contain '::cron::install'
+  contain '::cron::service'
 
-  include ::cron::install
-  include ::cron::service
+  Class['cron::install'] -> Class['cron::service']
+
+  # Manage cron.allow and cron.deny
+  if $manage_users_allow {
+    file { '/etc/cron.allow':
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      content => epp('cron/users.epp', { 'users' => $users_allow }),
+    }
+  }
+
+  if $manage_users_deny {
+    file { '/etc/cron.deny':
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      content => epp('cron/users.epp', { 'users' => $users_deny }),
+    }
+  }
 
-  anchor { 'cron::start': }
-  -> Class['cron::install']
-  -> Class['cron::service']
-  -> anchor { 'cron::end': }
 
   # Create jobs from hiera
-  $cron_job = hiera_hash('cron::job', undef)
-  if $cron_job {
-    create_resources('cron::job',$cron_job)
+
+  $cron_job = lookup('cron::job', Optional[Hash], 'hash', {})
+  $cron_job.each | String $t, Hash $params | {
+    cron::job { $t:
+      * => $params
+    }
   }
 
-  $cron_job_multiple = hiera_hash('cron::job::multiple', undef)
-  if $cron_job_multiple {
-    create_resources('cron::job::multiple', $cron_job_multiple)
+  $cron_job_multiple = lookup('cron::job::multiple', Optional[Hash], 'hash', {})
+  $cron_job_multiple.each | String $t, Hash $params | {
+    cron::job::multiple { $t:
+      * => $params
+    }
   }
 
-  $cron_job_hourly = hiera_hash('cron::hourly', undef)
-  if $cron_job_hourly {
-    create_resources('cron::hourly', $cron_job_hourly)
+  $cron_job_hourly = lookup('cron::job::hourly', Optional[Hash], 'hash', {})
```

Comment:
@crayfishx Is this right?  The previous code did a lookup for `cron::hourly`, the replacement uses `cron::job::hourly`.